lib/cartopy/mpl/geoaxes.py

```diff
@@ -587,7 +586,18 @@ def cla(self):
         self.dataLim.intervalx = self.projection.x_limits
         self.dataLim.intervaly = self.projection.y_limits
 
-        return result
+    if mpl.__version__ >= '3.6':
+        def clear(self):
+            """Clear the current Axes and add boundary lines."""
+            result = super().clear()
+            self._clear()
+            return result
+    else:
+        def cla(self):
+            """Clear the current Axes and add boundary lines."""
+            result = super().cla()
+            self._clear()
+            return result
```

Comment:
Rather than explicit version gating, what about just defining clear ourselves even for older versions of MPL.

```suggestion
    def clear(self):
        """Clear the current Axes and add boundary lines."""
        result = getattr(super(), "clear", super().cla)()
        self._clear()
        return result

    def cla(self):
        """Clear the current Axes and add boundary lines."""
        return self.clear()
```

Reply:
`Axes.clear` always existed though, so I don't think that `getattr` fallback will work.

Reply:
Yep I agree, I don't think I realized that it was the coming back down calls that were being missed. It might be helpful to put a quick comment about removing this once we only require MPL 3.6.